thirdparty/kyaml/runfn/runfn.go

```diff
@@ -438,27 +443,30 @@ func (r *RunFns) defaultFnFilterProvider(spec runtimeutil.FunctionSpec, fnConfig
 				AllowMount: true,
 			},
 		}
-		cf := &runtimeutil.FunctionFilter{
+		fltr = &runtimeutil.FunctionFilter{
 			Run:            c.Run,
 			FunctionConfig: fnConfig,
 			DeferFailure:   spec.DeferFailure,
 			ResultsFile:    resultsFile,
 		}
-		return cf, nil
+		name = spec.Container.Image
 	}
 
 	if spec.Exec.Path != "" {
 		e := &fnruntime.ExecFn{
 			Path: spec.Exec.Path,
 		}
-		ef := &runtimeutil.FunctionFilter{
+		fltr = &runtimeutil.FunctionFilter{
 			Run:            e.Run,
 			FunctionConfig: fnConfig,
 			DeferFailure:   spec.DeferFailure,
 			ResultsFile:    resultsFile,
 		}
-		return ef, nil
+		name = spec.Exec.Path
 	}
-
-	return nil, nil
+	// if we write back to file system, we need to print progress
+	if r.Output == nil {
+		return fnruntime.NewFunctionRunner(r.Ctx, fltr, name)
```

Comment:
We should always be using `FunctionRunner`, that's the idea. Lets introduce a field `DisableOutput` or something in `FunctionRunner` and configure accordingly.

Reply:
done